synapse/rest/client/v1/room.py

```diff
@@ -230,15 +233,23 @@ async def on_POST(self, request, room_id, event_type, txn_id=None):
             "sender": requester.user.to_string(),
         }
 
-        if b"ts" in request.args and requester.app_service:
+        inherit_depth = False
+        if prev_events:
```

Comment:
Should the `prev_event` query param also require the client to be a `requester.app_service` like the `ts` query parameter already does?

Reply:
Added